File: docs/algorithm/code/1.两数之和/Code.java
import java.util.Map;
import java.util.HashMap;

class Code {
    public static void main(String[] args) {
        int[] nums = { 1, 2, 7, 4, 5, 6 };
        int target = 9;
        int[] temp = twoSum(nums, target);
        for (int i = 0; i < temp.length; i++) {
            System.out.print(temp[i] + ",");
        }
    }

    public static int[] twoSum(int[] nums, int target) {
        if (nums == null || nums.length < 2)
            return new int[] {};
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            if (map.containsKey(complement)) {
                return new int[] { map.get(complement), i };
            }
            map.put(nums[i], i);
        }
        return new int[] {};

    }
}

File: docs/algorithm/simple/code/1.两数之和/Code.dart
main() {
  print(twoSum([1, 2, 3, 4, 5, 7], 9));
}

List<int> twoSum(List<int> nums, int target) {
  if (nums == null || nums.length < 2) return [];
  for (int i = 0; i < nums.length; i++) {
    for (var j = i + 1; i < nums.length; i++) {
      if(target == nums[i] + nums[j]){
        return [i,j];
      }
    }
  }
  return [];
}

List<int> twoSum2(List<int> nums, int target) {
  if (nums == null || nums.length < 2) return [];
  Map<int, int> map = {};
  for (int i = 0; i < nums.length; i++) {
    int complement = target - nums[i];
    if (map.containsKey(complement)) {
      return [map[complement], i];
    }
    map[nums[i]] = i;
  }
  return [];
}
